packages/plugin-ext/src/hosted/node/plugin-host.ts

```diff
@@ -67,19 +71,37 @@ process.on('rejectionHandled', (promise: Promise<any>) => {
     }
 });
 
+let terminating = false;
 const emitter = new Emitter();
 const rpc = new RPCProtocolImpl({
     onMessage: emitter.event,
     send: (m: {}) => {
-        if (process.send) {
+        if (process.send && !terminating) {
             process.send(JSON.stringify(m));
         }
     }
 });
 
-process.on('message', (message: string) => {
+process.on('message', async (message: string) => {
+    if (terminating) {
+        return;
+    }
     try {
-        emitter.fire(JSON.parse(message));
+        const msg = JSON.parse(message);
+        if ('type' in msg && msg.type === MessageType.Terminate) {
+            terminating = true;
+            emitter.dispose();
+            await Promise.race([
+                pluginHostRPC.terminate(),
```

Comment:
Could we send order directly to pluginManager through RPC  getProxy(`MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT`) ?

Reply:
I thought it makes sense from lifecycle perspective to handle inside `PluginHostRPC`, since it is responsible to create the manager, it should be responsible to terminate it.

Reply:
yes make sense